### ISSUE TITLE
[Inferencing] How to initialize model before .predict()

### ISSUE THREAD

**Issue by longtran1904** [ultralytics/ultralytics#19803]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I'm using simpel model.predict() command to do inference.

However, I'm using Multiprocessing to split the dataset and try to inference in a concurrent way.

```
def process_safe_predict(batch_size, image_path, output_file):
    logger = get_thread_logger(output_file) # My custom logger for each process

    # Process results
    results = global_model.predict(image_path, imgsz=480, stream=True, batch=batch_size)
    total_time = 0
    count = 0
    for i, result in enumerate(results):  
        # Extract inference time
        inference_time = result.speed["inference"]  # In milliseconds (ms)
        preprocess_time = result.speed["preprocess"]  # Preprocessing time (ms)
        postprocess_time = result.speed["postprocess"]  # Postprocessing time (ms)
        
        total_time += inference_time
        count += 1
        
        logger.info(f"Image {i}: \nPreprocess time {preprocess_time} \nInference time {inference_time} ms\nPostprocess time {postprocess_time} ms")
        
    logger.info(f"Average Inferencing Time: {(total_time / count):.2f} ms")
```

I launch this function for each process
```
        process = multiprocessing.Process(target=process_safe_predict, args=(args.batch_size, output_folder+f'/part_{i+1}', inference_file))
```

I noticed that the command model.predict(...) initializes and fetches the actual model in `/engine/model.py`

```
if not self.predictor:
            self.predictor = (predictor or self._smart_load("predictor"))(overrides=args, _callbacks=self.callbacks)
            self.predictor.setup_model(model=self.model, verbose=is_cli)
        else:  # only update args if predictor is already setup
            self.predictor.args = get_cfg(self.predictor.args, args)
            if "project" in args or "name" in args:
                self.predictor.save_dir = get_save_dir(self.predictor.args)
        if prompts and hasattr(self.predictor, "set_prompts"):  # for SAM-type models
            self.predictor.set_prompts(prompts)
        return self.predictor.predict_cli(source=source) if is_cli else self.predictor(source=source, stream=stream)
```

Is there any way I can load the model before calling model.predict()?
I would like to load first so that processes can share the same model on memory to do inference

**Comment by UltralyticsAssistant** [comment]
👋 Hello @longtran1904, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it effectively.

If this is a custom training ❓ Question or implementation inquiry, please provide as much relevant information as possible. Details such as logs, dataset examples, and your environment setup are helpful for us to better assist you. Additionally, for optimal results, you can review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

To address your inquiry:
- Ensure you are using the most up-to-date version of the `ultralytics` package. You can upgrade with:
```bash
pip install -U ultralytics
```

- For multiprocessing setups with shared models, we recommend checking our [Tasks documentation](https://docs.ultralytics.com/tasks/) and [Modes documentation](https://docs.ultralytics.com/modes/) for insights into streamlining inference workflows.

Join the Ultralytics community where it suits you best. For real-time chat, join us on [Discord](https://discord.com/invite/ultralytics) 🎧. For in-depth discussions, explore [Discourse](https://community.ultralytics.com/). Or dive into our [Subreddit](https://reddit.com/r/Ultralytics) to exchange ideas with fellow users.

## Environments

YOLO can be run in any of the following verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

📌 This is an automated response, but an Ultralytics engineer will review your issue and provide further assistance soon. Thank you for your patience! 😊

**Comment by Y-T-G** [comment]
There's no easy way to initialize the model without running prediction.

I also don't think this would make inference faster since torch, numpy and cv2 all use multi-threading by default. It would only add unnecessary overhead.

If you want to utilize multiple CPUs, just use `cv2.setNumThreads()`, `torch.set_num_threads()` and os.environ["OMP_NUM_THREADS"] to set the threads.
